src/System.CommandLine/Parsing/CommandResult.cs

```diff
@@ -125,7 +125,7 @@ private void ValidateOptions(bool completeValidation)
                 // When_there_is_an_arity_error_then_further_errors_are_not_reported
                 if (!ArgumentArity.Validate(argumentResult, out var error))
                 {
-                    optionResult.AddError(error.ErrorMessage!);
+                    argumentResult.AddError(error.ErrorMessage!);
```

Comment:
I do not understand this change. If this is on options, I thought we got the errors from the option. Do we get them from the option's argument?

Reply:
I want to discuss the design here. I do not think the Diagram subsystem should know that `CliOption` treats its argument as a `CliArgument`. My intuition is that this problem will appear in numerous places. 

I'd like to discuss this prior to merging.